source/multigrid/mg_transfer_prebuilt.cc

```diff
@@ -284,23 +284,43 @@ void MGTransferPrebuilt<VECTOR>::build_matrices (
   // We keep track in the bitfield dof_touched which global dof has
   // been processed already (on the current level). This is the same as
   // the multigrid running in serial.
-  // Only entering on the finest level gives wrong results (why?)
+
+  struct dof_pair
+  {
+    unsigned int level;
+    types::global_dof_index global_dof_index;
+    types::global_dof_index level_dof_index;
+
+    dof_pair(unsigned int level,
+             types::global_dof_index global_dof_index,
+             types::global_dof_index level_dof_index)
+      :
+      level(level), global_dof_index(global_dof_index), level_dof_index(level_dof_index)
+    {}
+
+    dof_pair()
+    {}
```

Comment:
I do, otherwise I can not put them into a vector and resize it before calling MPI_recv.